ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx

```diff
@@ -17,65 +22,180 @@ import {
+        (account) => account !== address,
+      );
+    }
+    setSelectedAccounts(newSelectedAccounts);
+  };
+
+  const disconnectAllAccounts = () => {
+    const subject = (subjects as SubjectsType)[activeTabOrigin];
+    if (subject) {
+      const permissionMethodNames = Object.values(subject.permissions).map(
+        ({ parentCapability }: { parentCapability: string }) =>
+          parentCapability,
+      ) as string[];
+      if (permissionMethodNames.length > 0) {
+        const permissionsRecord: Record<string, string[]> = {
+          [activeTabOrigin]: permissionMethodNames,
+        };
+
+        dispatch(
+          removePermissionsFor(
+            permissionsRecord as Record<string, NonEmptyArray<string>>,
+          ),
+        );
+      }
+    }
+  };
+
+  const managePermittedAccounts = (
+    selectedAccounts: string[],
+    connectedAccountsAddresses: string[],
+    activeTabOrigin: string,
+  ) => {
+    const removedElements = connectedAccountsAddresses.filter(
+      (account) => !selectedAccounts.includes(account),
+    );
+
+    if (removedElements.length > 0) {
+      removedElements.forEach((account) => {
+        dispatch(removePermittedAccount(activeTabOrigin, account));
+      });
+    }
+
+    const newElements = selectedAccounts.filter(
+      (account) => !connectedAccountsAddresses.includes(account),
+    );
+
+    if (newElements.length > 0) {
+      dispatch(addMorePermittedAccounts(activeTabOrigin, newElements));
+    }
+  };
 
   return (
     <Modal
       isOpen
-      onClose={onClose} // Simplified inline function
+      onClose={onClose}
       data-testid="edit-accounts-modal"
       className="edit-accounts-modal"
     >
       <ModalOverlay />
       <ModalContent>
         <ModalHeader onClose={onClose}>{t('editAccounts')}</ModalHeader>
         <Box padding={4}>
-          <Checkbox
-            label={t('selectAll')}
-            isChecked
-            gap={4}
-            // Uncomment and implement these if needed:
-            // onClick={() => (allAreSelected() ? deselectAll() : selectAll())}
-            // isIndeterminate={isIndeterminate}
-          />
+          <Checkbox label={t('selectAll')} isChecked gap={4} />
         </Box>
         {mergedAccounts.map((account) => (
           <AccountListItem
+            onClick={() => handleAccountClick(account.address)}
             account={account}
             key={account.address}
             isPinned={Boolean(account.pinned)}
-            startAccessory={<Checkbox isChecked />}
-            onClick={() => console.log(null)}
+            startAccessory={
+              <Checkbox
+                isChecked={selectedAccounts.includes(account.address)}
+              />
+            }
             selected={false}
           />
         ))}
+
+        <ModalFooter>
+          {selectedAccounts.length === 0 ? (
+            <ButtonPrimary
+              data-testid="connect-more-accounts-button"
```

Comment:
Would it be helpful to have a separate value here?  `disconnect-more-accounts-button`?

Reply:
Updated it